test/checkpoint/checkpointer.c

```diff
@@ -283,11 +258,14 @@ real_checkpointer(void)
               timestamp_buf, sizeof(timestamp_buf), "oldest_timestamp=%" PRIx64, g.ts_oldest));
             testutil_check(g.conn->set_timestamp(g.conn, timestamp_buf));
         }
-        /* Random value between 4 and 8 seconds. */
-        if (g.sweep_stress) {
-            delay = __wt_random(&rnd) % 5;
-            __wt_sleep(delay + 4, 0);
-        }
+
+        if (g.sweep_stress)
+            /* Random value between 4 and 8 seconds. */
+            delay = __wt_random(&rnd) % 5 + 4;
+        else
+            /* Just find out if we should flush_tier. */
+            delay = 0;
+        testutil_tiered_sleep(&g.opts, delay, &flush_tier);
```

Comment:
I don't understand the intent of `sweep_stress`. Is it important that we wait 4-8 seconds between checkpoints? Because if tiered storage is enabled, we will never sleep that long because of the much shorter delay between flushes.

Reply:
Yes, I agree this should be changed.  I think `sweep_stress` means longer time (multiple seconds) between checkpoints, so each checkpoint is more stressful.  If that variable is not on, then we checkpoint in a tight loop.  This is unchanged behavior from the previous version (although the code got moved around).  The previous version also had a flush_tier thread that sleeps < 10000 usec.  Now, the flush is done by calling checkpoint("flush_tier=(enabled)") and that incorporates a checkpoint, of course.

I think the right thing, and in line with original intention is to keep this code (and thus the checkpoint interval behavior) the same, but make the flush_tier interval also look at `g.sweep_stress`.  If stress is off, interval is < 10000 usecs, as we have now. If stress is on, flush_tier interval is, say, between 5 and 15 seconds, that is, generally (but not always) somewhat longer than the checkpoint rate.

There is a question is how long the runs in test/checkpoint last (a 10 second run would not give us many flush calls).  It does not run on a timer, but rather completes when each of _W_ workers finishes _n_ operations.  For evergreen runs with `-s 1` (that sets `g.sweep_stress`), _W_ is 3 and _n_ is 100000.  I observe this runs ~45 seconds when run with timestamps, and ~90 seconds without timestamps.  So a flush_tier between 5 and 15 seconds should give something like 4 to 5 flush_tiers for these stress tests with timestamps, and ~9 flush_tiers without timestamps.  So that feels adequate, although when processors get faster, we may need to recalibrate (perhaps by doing more operations per run).

@sueloverso I expect you know test/checkpoint pretty well, can you comment on the above - I can brief you on what we're doing if it helps.